lib/iris/_lazy_data.py

```diff
@@ -126,6 +126,34 @@ def as_concrete_data(data):
     return data
 
 
+def lazy_masked_fill_value(data):
+    """
+    Return the fill value of a lazy masked array.
+
+    Args:
+
+    * data:
+        A dask array, NumPy `ndarray` or masked array
+
+    Returns:
+        The fill value of `data` if `data` represents a masked array, or None.
+
+    """
+    # If lazy, get the smallest slice of the data from which we can retrieve
+    # the fill_value.
+    if is_lazy_data(data):
+        inds = tuple([0] * (data.ndim-1))
+        smallest_slice = data[inds][:0]
+        data = as_concrete_data(smallest_slice)
```

Comment:
Should work; see [these tests](https://github.com/dkillick/iris/blob/68e2b6676d436728d3ca6b666bf356ad556e6b7d/lib/iris/tests/unit/lazy_data/test_lazy_masked_fill_value.py#L48-L56)... although I'm not convinced that covers the `MaskedConstant` case. Thoughts?